packages/marshal/src/marshal.js

```diff
@@ -538,10 +538,10 @@ export function makeMarshal(
   );
   // Ascending numbers identifying the sending of errors relative to this
   // marshal instance.
-  let errorCount = 0;
+  let errorIdNum = 10000;
```

Comment:
I'm -0 on the change to the starting value. I see three benefits to doing something other than count-up-from-0:

* avoid visual collision with values created by other instances (the "good grief, everything on the screen is labeled `#1`" problem)
* avoid revealing private information (like how many errors have been raised/received) to someone who can observe these numbers
* avoid false-positives in unit tests that assert an observed number is "1" or "2" but are tricked by a count from some entirely different category

To avoid revealing private information, these need to be random or pseudo-random, so merely changing the starting offset won't change the fact that it's still a counter.

To address the other two collision problems require that each instance/category use a different starting point. For Swingset, we allocate object-ids and promise-ids with different starting values, but all vats use the same (because we haven't yet added a coordination/configuration mechanism to tell them to do anything differently). So we've only half-addressed the problem, and it's still awfully hard to visually read a trace that includes messages from more than one vat at a time.

The more complete fix is either to pass in some distinctive starting point, or make sure that wherever this counter is displayed, it is printed next to a distinctive identifier for the next level up the hierarchy. E.g. `makeMarshal()` could be given an argument to use as a distinctive string. Then we could have `errorCount` still be a counter (modulo the private-information thing) and start at 0, and still get something distinctive to look at. And when we read the output, we can talk about "the fifth Error unserialized by Marshal 37", instead of trying to glean the which-Marshal-is-it by human-parsing a bigger number.

I.e.: lean into the hierarchy, defer the private-information fix.

That said, it's a minor thing, I'd be happy to have this land without it.

Reply:
Ok, I leaned in a bit more but still far from what's needed. Part of your suggestion was already there: the caller of `makeMarshal` can and usually does provide a `marshalName` option that serves as the base of the hierarchy, whatever that level is according to the caller. I brought out the starting `errorIdNum` as another `makeMarshal` option. I then went to each of the call sites and gave each appropriate one its own distinct starting `errorIdNum`, which should still help users some in distinguishing these. I marked all the relevant site with a pointer to #2780